test/__init__.py

```diff
@@ -1,5 +1,38 @@
 from __future__ import print_function, division, absolute_import
 
+import os
+from typing import Callable, List
 
 from . import rhsm_display
 rhsm_display.set_display()
+
+import pytest
+
+subman_marker_dbus = pytest.mark.dbus
+subman_marker_functional = pytest.mark.functional
+subman_marker_slow = pytest.mark.slow
+# This allows us to set higher timeout limit for tests that are known to be slow
+subman_marker_slow_timeout = pytest.mark.timeout(40)
+
+
+def subman_marker_zypper(func: Callable) -> Callable:
+    """Do not run zypper tests even if the zypper marker is set.
+
+    Because these tests require environment variables, this additional
+    check makes sure the test suite won't error out when you overwrite
+    the defaults in CLI, e.g. with '-m "not dbus"'.
+    """
+    missing_vars: List[str] = [
+        v
+        for v
+        in ['RHSM_USER', 'RHSM_PASSWORD', 'RHSM_URL', 'RHSM_POOL', 'RHSM_TEST_REPO', 'RHSM_TEST_PACKAGE']
+        if v not in os.environ
+    ]
+
+    skip_func: Callable = pytest.mark.skipif(
+        missing_vars,
+        reason=f"Missing Zypper environment variables {', '.join(missing_vars)}."
+    )
+    mark_func: Callable = pytest.mark.zypper
+
+    return skip_func(mark_func(func))
```

Comment:
while the idea of this is great, I fear it is too specific and will be easily outdated

since the test this applies to requires environment variables, I'd rather add a `skipifnotenvvar('ENV_VAR')` function/decorator that skips a test if an environment variable is not set:
- we can add all the ones needed to the zypper test (pytests checks all the skip checks decorated to a function)
- it can possibly be used in other tests if needed
- it should be smaller and easier to maintain

having this kind of checks done as decorator would allow to drop almost all the content of `TestServicePlugin.setUp()`

Reply:
Done.